### ISSUE TITLE
IMAP: AndroidRuntime: java.lang.RuntimeException: com.fsck.k9.mail.MessagingException: LocalFolder.open(): Folder not found: null (6)

### ISSUE THREAD

**Issue by winny-** [k9mail/k-9#4903]
Related #4848. Unlike the linked issue, I didn't migrate phones or use a backup/restore tool. It's the original install of K-9 mail upgraded to the beta release.

### Expected behavior
Switch to account mailbox.

### Actual behavior
Crashes on selecting IMAP3 from account switcher.

### Steps to reproduce
1. Touch hamburger top left (Accounts)
2. Touch affected account's icon

### Environment
Installed from f-droid.

K-9 Mail version: 5.717

Android version: 9

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors) and attach them here:

[k9-crash.txt](https://github.com/k9mail/k-9/files/5043734/k9-crash.txt) (Gisted copy https://gist.github.com/eff9934856d98b52787f9f3d9f99f16d )

#### Backtrace
(for search indexing)

```
08-07 16:24:25.474 30157 30157 E AndroidRuntime: java.lang.RuntimeException: com.fsck.k9.mail.MessagingException: LocalFolder.open(): Folder not found: null (6)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.fragment.MessageListFragment.getFolderInfoHolder(MessageListFragment.java:524)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.fragment.MessageListFragment.decodeArguments(MessageListFragment.java:468)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.fragment.MessageListFragment.onCreate(MessageListFragment.java:317)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2684)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:280)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1175)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentTransition.addToFirstInLastOut(FragmentTransition.java:1255)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentTransition.calculateFragments(FragmentTransition.java:1138)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentTransition.startTransitions(FragmentTransition.java:136)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1989)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentController.execPendingActions(FragmentController.java:447)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at androidx.fragment.app.FragmentActivity.onResume(FragmentActivity.java:458)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.ui.base.K9Activity.onResume(K9Activity.kt:21)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.activity.MessageList.onResume(MessageList.java:553)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1434)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.Activity.performResume(Activity.java:7304)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.ActivityThread.performNewIntents(ActivityThread.java:3352)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:3367)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:49)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1986)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:215)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6939)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
08-07 16:24:25.474 30157 30157 E AndroidRuntime: Caused by: com.fsck.k9.mail.MessagingException: LocalFolder.open(): Folder not found: null (6)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.mailstore.LocalFolder$1.doDbWork(LocalFolder.java:178)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.mailstore.LocalFolder$1.doDbWork(LocalFolder.java:160)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.mailstore.LockableDatabase.execute(LockableDatabase.java:273)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.mailstore.LocalFolder.open(LocalFolder.java:160)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.fragment.MlfUtils.getOpenFolder(MlfUtils.java:28)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        at com.fsck.k9.fragment.MessageListFragment.getFolderInfoHolder(MessageListFragment.java:521)
08-07 16:24:25.474 30157 30157 E AndroidRuntime:        ... 29 more

```

**Comment by winny-** [comment]
I attempted to fire up K-9 in android studio and reproduce the bug.

Unfortunately while attempting to modify the application ID so that K-9 Mail can use my existing configuration, the configuration was erased. I think when I removed the app to install the other app, configuration was deleted as per [`android:allowBackup="false"`](https://developer.android.com/guide/topics/data/autobackup).

I reinstalled the f-droid build (5.717) then imported my settings via the *Welcome screen* and have not been able to reproduce the crash since the aforementioned accidental configuration erasure.

Closing because I can't reproduce the crash.

**It might be a good idea to document how to deploy K-9 Mail in place over `com.fsck.k9` so that weird edge case bugs can be debugged on the device they occur on, because let's face it everyone's android is a little unique in some way or another. :)**
